File: PERIOD_1/p1_week1/cash.c
#include <cs50.h>       // For get_int
#include <stdio.h>      // For printf

// Function prototypes for coin calculations
int calculate_quarters(int cents);
int calculate_dimes(int cents);
int calculate_nickels(int cents);
int calculate_pennies(int cents);

int main(void)
{
    int cents;          // Variable to store change owed
    do
    {
        cents = get_int("Change owed: ");
    }
    while (cents < 0);  // Ensure non-negative input

    // Calculate number of coins
    int quarters = calculate_quarters(cents);
    cents = cents - quarters * 25;      // how many quarters fit

    int dimes = calculate_dimes(cents);
    cents = cents - dimes * 10;         // remaining how many dimes fit

    int nickels = calculate_nickels(cents);
    cents = cents - nickels * 5;        // remaining how many nickels fit

    int pennies = calculate_pennies(cents);
    cents = cents - pennies * 1;        // remaining how many pennies fit until 0

    // Sum coins
    int coins = quarters + dimes + nickels + pennies;

    // Print result
    printf("%i\n", coins);              // % placeholder for an integer
}

// Function definitions
int calculate_quarters(int cents)       // each function takes the amount left in cents and divides it by the coin value
{
    return cents / 25;
}

int calculate_dimes(int cents)
{
    return cents / 10;
}

int calculate_nickels(int cents)
{
    return cents / 5;
}

int calculate_pennies(int cents)
{
    return cents / 1;
}

File: PERIOD_2/p2_week1/sentimental-cash/cash.py
from cs50 import get_float

change = 0                                       # variable to store the amount of change the user enters
coins = 0                                        # to keep track of the total number of coins used

while change <= 0:                               # Prompt user for change (in dollars)
    change = get_float("Change: ")

def remain(coin_value):                          # Function to count how many coins of a given value can be used
    global change                                # call the change function
    global coins
    while (change >= coin_value):
        change = round(change - coin_value, 10)  # subtract coin and round to avoid floating errors
        coins += 1                               # count each coin used


# Try each coin type
remain(0.25)    # quarters
remain(0.10)    # dimes
remain(0.05)    # nickels
remain(0.01)    # pennies

# Print total coins used
print(coins)
